core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java

```diff
@@ -319,12 +320,24 @@ private void deleteTableMetadata(String namespace, String table) throws Executio
 
   @Override
   public void dropNamespace(String namespace) throws ExecutionException {
-    if (!databaseExists(namespace)) {
-      throw new ExecutionException("the database does not exist");
-    }
-
     try {
       client.getDatabase(namespace).delete();
+      CosmosContainer namespacesContainer = getNamespacesContainer();
+      namespacesContainer.deleteItem(
+          new CosmosNamespace(namespace), new CosmosItemRequestOptions());
+
+      // Delete the namespaces container and metadata database if there is no more existing
+      // namespaces
+      if (!namespacesContainer
+          .queryItems(
+              "SELECT 1 FROM container OFFSET 0 LIMIT 1",
+              new CosmosQueryRequestOptions(),
+              Object.class)
+          .stream()
+          .findFirst()
+          .isPresent()) {
+        client.getDatabase(metadataDatabase).delete();
```

Comment:
Don't we need to delete the namespaces container before deleting the metadata database?

Reply:
Deleting the database will delete all its container so deleting the container first is not required.